source/common/http/header_map_impl.cc

```diff
@@ -324,28 +319,24 @@ HeaderMapImpl::HeaderMapImpl(
 
 void HeaderMapImpl::updateSize(uint64_t from_size, uint64_t to_size) {
   // Removes from_size from cached_byte_size_ and adds to_size.
-  if (cached_byte_size_.has_value()) {
-    ASSERT(cached_byte_size_ >= from_size);
-    cached_byte_size_.value() -= from_size;
-    cached_byte_size_.value() += to_size;
-  }
+  ASSERT(cached_byte_size_ >= from_size);
+  cached_byte_size_ -= from_size;
+  cached_byte_size_ += to_size;
 }
 
 void HeaderMapImpl::addSize(uint64_t size) {
-  // Adds size to cached_byte_size_ if it exists.
-  if (cached_byte_size_.has_value()) {
-    cached_byte_size_.value() += size;
-  }
+  // Adds size to cached_byte_size_.
+  cached_byte_size_ += size;
 }
 
 void HeaderMapImpl::subtractSize(uint64_t size) {
-  if (cached_byte_size_.has_value()) {
-    ASSERT(cached_byte_size_ >= size);
-    cached_byte_size_.value() -= size;
-  }
+  ASSERT(cached_byte_size_ >= size);
+  cached_byte_size_ -= size;
 }
 
 void HeaderMapImpl::copyFrom(const HeaderMap& header_map) {
+  // Clear the HeaderMap before copying the headers.
```

Comment:
This seems like the right thing to do, but this seems like a behavior change from what we did before? WDYT?

Reply:
Hmmm without it I broke a test in one of the commits I pushed after the move to string_views since `addViaMove` called in the copy constructor will append to the inline headers already set in `request_headers_`... 
https://github.com/envoyproxy/envoy/blob/fb628ce25441d5f7d424fee17f9d773c335bc152/test/common/access_log/access_log_impl_test.cc#L382
I have no clue why this isn't an issue anymore... will spend some time looking at it.